prow/github/client.go

```diff
@@ -519,6 +521,82 @@ func (c *Client) UnassignIssue(org, repo string, number int, logins []string) er
 	return nil
 }
 
+func (c *Client) tryRequestPRReview(org, repo string, number int, logins []string) (int, error) {
+	c.log("RequestPRReview", org, repo, number, logins)
+	var pr PullRequest
+	return c.request(&request{
+		method:      http.MethodPost,
+		path:        fmt.Sprintf("%s/repos/%s/%s/pulls/%d/requested_reviewers", c.base, org, repo, number),
+		requestBody: map[string][]string{"reviewers": logins},
+		exitCodes:   []int{201},
+	}, &pr)
+}
+
+// RequestPRReview tries to add the users listed in 'logins' as requested reviewers of the specified PR.
+// If any user in the 'logins' slice is not a contributor of the repo, the entire POST will fail
+// without adding any reviewers. The github API response does not specify which user(s) were invalid
+// so if we fail to request reviews from the members of 'logins' we try to request reviews from
+// each member individually. We try first with all users in 'logins' for efficiency in the common case.
+func (c *Client) RequestPRReview(org, repo string, number int, logins []string) error {
+	statusCode, err := c.tryRequestPRReview(org, repo, number, logins)
+	if err != nil && statusCode == 422 {
+		//failed to set all members of 'logins' as reviewers, try individually.
+		var missing MissingUsers
+		for _, user := range logins {
+			statusCode, err = c.tryRequestPRReview(org, repo, number, []string{user})
+			if err != nil && statusCode == 422 {
+				//user is not a contributor
+				missing = append(missing, user)
+			} else if err != nil {
+				return fmt.Errorf("Unexpected error or status code while Requesting PR Review. Status: %d, errmsg: %v", statusCode, err)
```

Comment:
Error strings should not be capitalized. This one also probably doesn't need to be so verbose.